### DISCUSSION TITLE
How to retrieve value of a key without having to write the entire key?

### DISCUSSION THREAD

**Discussion by dev-vaayen** [etcd-io/etcd#16334] [Q&A] [answered]
Let's say I have a key which stores information about some children who study at a school. The value is going to be the name of the child

The key looks like:
/schools/schoolName/class/rollNumber
The value is going to be:
Child's_Name

**Example**
Key:
/schools/ColomboSchool/5th/rollNumber 
Value:
Mallinga

My question is: How can I retrieve the name of ALL children who are studying in the ColomboSchool, whose roll number is 22? Let's just assume I do not know or I do not want to enter the class/grade and I want to see **all the children** that study at ColomboSchool & have roll number 22.

Is there any pre-built command in etcdctl to make this happen? Or how can I find the names of all students studying in ColomboSchool without mentioning their class/grade or rollNumber?

I did use `etcdctl get --prefix "/schools/" | grep "ColomboSchool" | grep "22"` and was able to retrieve the relevant data. Is there any other way to retrieve data?

**Comment by jmhbnz** [comment, marked as answer]
Hey @Anand11Yash - Thanks for your question.  etcd as a key value store doesn't have complex query capabilities like a traditional relational database, so you need to structure your keys in a way that makes querying feasible.

The approach you are taking with `etcdctl get --prefix` and then further filter with additional tools like `grep` makes sense to me. Perhaps just extend your prefix to also include `ColomboSchool` so one less grep iteration is required.

```bash
etcdctl get --prefix "/schools/ColomboSchool" | grep "22"
```
